test/support/src/array_templates.h

```diff
@@ -1379,8 +1538,125 @@ uint64_t num_cells(const F& fragment, const auto& field_sizes) {
   }(std::tuple_cat(fragment.dimensions(), fragment.attributes()));
 }
 
+/**
+ * Writes a fragment to an array.
+ */
+template <typename Asserter, FragmentType Fragment>
+void write_fragment(
+    const Fragment& fragment,
+    Array& forwrite,
+    tiledb_layout_t layout = TILEDB_UNORDERED) {
+  Query query(forwrite.context(), forwrite, TILEDB_WRITE);
```

Comment:
Instead of a context accessor in the `Array` class, we should add a constructor overload in `Query` that does not explicitly accept a context, and gets it from the array; [C# already has it](https://github.com/TileDB-Inc/TileDB-CSharp/blob/1c8e0e75b7aba5a28d0f08d5075b02f3b3572ced/sources/TileDB.CSharp/Query.cs#L111-L122) for example.